server/src/main/scala/actors/FixedPointsActorBase.scala

```diff
@@ -52,42 +53,44 @@ class FixedPointsActorBase extends RecoveryActorLike with PersistentDrtActor[Fix
 
   override def persistenceId = "fixedPoints-store"
 
-  var state = initialState
+  var state: FixedPointsState = initialState
 
-  def initialState = FixedPointsState("")
+  def initialState = FixedPointsState(StaffAssignmentHelper.empty)
 
   val snapshotInterval = 1
   override val snapshotBytesThreshold: Int = oneMegaByte
 
   import FixedPointsMessageParser._
 
-  override def stateToMessage: GeneratedMessage = FixedPointsStateSnapshotMessage(fixedPointsStringToFixedPointsMessages(state.fixedPoints))
+  override def stateToMessage: GeneratedMessage = FixedPointsStateSnapshotMessage(staffAssignmentsToFixedPointsMessages(state.staffAssignments))
 
   def processSnapshotMessage: PartialFunction[Any, Unit] = {
-    case snapshot: FixedPointsStateSnapshotMessage => state = FixedPointsState(fixedPointMessagesToFixedPointsString(snapshot.fixedPoints.toList))
+    case snapshot: FixedPointsStateSnapshotMessage => state = FixedPointsState(fixedPointMessagesToStaffAssignments(snapshot.fixedPoints))
   }
 
   def processRecoveryMessage: PartialFunction[Any, Unit] = {
     case fixedPointsMessage: FixedPointsMessage =>
-      val fp = fixedPointMessagesToFixedPointsString(fixedPointsMessage.fixedPoints.toList)
+      val fp = fixedPointMessagesToStaffAssignments(fixedPointsMessage.fixedPoints)
       updateState(fp)
   }
 
   def receiveCommand: Receive = {
     case GetState =>
       log.info(s"GetState received")
-      sender() ! state.fixedPoints
+      sender() ! state.staffAssignments
 
-    case fp: String if fp != state.fixedPoints =>
-      updateState(fp)
-      onUpdateState(fp)
+    case staffAssignments: StaffAssignments if staffAssignments != state.staffAssignments =>
+      updateState(staffAssignments)
+      onUpdateState(staffAssignments)
 
       log.info(s"Fixed points updated. Saving snapshot")
-      val snapshotMessage = FixedPointsStateSnapshotMessage(fixedPointsStringToFixedPointsMessages(state.fixedPoints))
+      val snapshotMessage = FixedPointsStateSnapshotMessage(staffAssignmentsToFixedPointsMessages(state.staffAssignments))
       saveSnapshot(snapshotMessage)
 
-    case _: String =>
+    case staffAssignments: StaffAssignments if staffAssignments == state.staffAssignments =>
```

Comment:
perhaps we should call these fixedPoints to be more clear about what is happening here.

Reply:
sure prob sensible